src/Aspire.Hosting/ApplicationModel/IValueProvider.cs

```diff
@@ -4,9 +4,9 @@
 namespace Aspire.Hosting.ApplicationModel;
 
 /// <summary>
-/// A interface that allows the value to be provided for an environment variable.
+/// An interface that allows the value to be provided for an environment variable.
 /// </summary>
-public interface IValueProvider
+public interface IValueProvider : IManifestExpressionProvider
```

Comment:
The thinking here is that we always implement both of these together. Maybe we should just combine them into 1 interface?

My other thought was to introduce a new interface that inherits from both, but I couldn't come up with a good name. `IValueReference` is all that I got, and I thought "why not just reuse IValueProvider?"

Reply:
It does look like everywhere we implement one, we implement the other (and vice versa).

Reply:
I don't think we want this but I understand why it seems attractive.

Reply:
Nested/recursive ReferenceExpressions then?